src/components/screens/SendScreen/helpers/index.tsx

```diff
@@ -0,0 +1,130 @@
+import Blockaid from "@blockaid/client";
+import useAppTranslation from "hooks/useAppTranslation";
+import { useMemo } from "react";
+import { SecurityLevel } from "services/blockaid/constants";
+import {
+  assessTransactionSecurity,
+  extractSecurityWarnings,
+} from "services/blockaid/helper";
+
+/**
+ * Hook for managing transaction security assessment
+ * Extracts security warnings and determines severity level
+ */
+export function useTransactionSecurity(
+  transactionScanResult: Blockaid.StellarTransactionScanResponse | undefined,
+) {
+  const transactionSecurityAssessment = useMemo(
+    () => assessTransactionSecurity(transactionScanResult),
+    [transactionScanResult],
+  );
+
+  const transactionSecurityWarnings = useMemo(() => {
+    if (
+      transactionSecurityAssessment.isMalicious ||
+      transactionSecurityAssessment.isSuspicious
+    ) {
+      const warnings = extractSecurityWarnings(transactionScanResult);
+
+      if (Array.isArray(warnings) && warnings.length > 0) {
+        return warnings;
+      }
+    }
+
+    return [];
+  }, [
+    transactionSecurityAssessment.isMalicious,
+    transactionSecurityAssessment.isSuspicious,
+    transactionScanResult,
+  ]);
+
+  const transactionSecuritySeverity:
+    | Exclude<SecurityLevel, SecurityLevel.SAFE>
+    | undefined = useMemo(() => {
+    if (transactionSecurityAssessment.isMalicious)
+      return SecurityLevel.MALICIOUS;
+    if (transactionSecurityAssessment.isSuspicious)
+      return SecurityLevel.SUSPICIOUS;
+
+    return undefined;
+  }, [
+    transactionSecurityAssessment.isMalicious,
+    transactionSecurityAssessment.isSuspicious,
+  ]);
+
+  return {
+    transactionSecurityAssessment,
+    transactionSecurityWarnings,
+    transactionSecuritySeverity,
+  };
+}
+
+export interface BannerContent {
+  text: string;
+  variant: "error" | "warning";
+  onPress: () => void;
+}
+
+export interface UseSendBannerContentParams {
+  isMalicious: boolean;
+  isSuspicious: boolean;
+  isRequiredMemoMissing?: boolean;
+  onSecurityWarningPress: () => void;
+  onMemoMissingPress?: () => void;
+}
+
+/**
+ * Hook for building banner content based on transaction security and memo requirements
+ * Returns banner configuration with text, variant, and onPress handler
+ */
+export function useSendBannerContent({
+  isMalicious,
+  isSuspicious,
+  isRequiredMemoMissing = false,
+  onSecurityWarningPress,
+  onMemoMissingPress,
+}: UseSendBannerContentParams): BannerContent | undefined {
+  const { t } = useAppTranslation();
+
+  return useMemo(() => {
+    const shouldShowNoticeBanner =
+      isRequiredMemoMissing || isMalicious || isSuspicious;
+
+    if (!shouldShowNoticeBanner) {
+      return undefined;
```

Comment:
this is so NIT, but I feel `null` is perhaps a better return value as `null` generally implies that this is intentionally set as empty (which is what we're doing here) where as `undefined` generally implies we've yet to set a value